src/features/Instructors/AddInstructors/index.jsx

```diff
@@ -99,12 +101,41 @@ const AddInstructors = ({ isOpen, onClose }) => {
     return null;
   };
 
+  const handleEditInstructor = async (e) => {
+    e.preventDefault();
+    try {
+      const formData = new FormData();
+      formData.append('enrollment_privilege', formState.instructor.hasEnrollmentPrivilege);
+      formData.append('institution_id', selectedInstitution.id);
+      formData.append('instructor_id', instructorInfo.instructorId);
+
+      await dispatch(editInstructor(formData));
+      handleCloseModal();
+      dispatch(updateInstructorAdditionRequest({ status: RequestStatus.INITIAL }));
+    } catch (error) {
+      logError(error);
+    }
+  };
+
   useEffect(() => {
     dispatch(updateInstructorAdditionRequest({ status: RequestStatus.INITIAL }));
 
     return () => dispatch(resetInstructorAdditionRequest());
   }, [dispatch]);
 
+  useEffect(() => {
+    if (isEditing && instructorInfo) {
+      setFormState(prev => ({
+        ...prev,
+        instructor: {
+          ...prev.instructor,
+          email: instructorInfo.instructorEmail,
+          hasEnrollmentPrivilege: instructorInfo?.hasEnrollmentPrivilege,
+        },
+      }));
+    }
+  }, [isEditing, instructorInfo]);
```

Comment:
Also another issue with `instructorInfo` this is being a memory leak, once the table change, for whatever the reason this state is being triggered, I did this small experiment to prove my theory:

```javascript
// src/features/Instructors/AddInstructors/index.jsx
useEffect(() => {
    console.log({ instructorInfo });
  }, [instructorInfo]);
``` 

If you sort the table using whatever column, like `Status` this state will be triggered once per each table record:

![image](https://github.com/user-attachments/assets/af44e38b-5acb-4737-8f24-c39c3106a2d7)

Reply:
I reviewed and happens in all tables so is not part of this scope, could you please add a ticket in the backlog as a technical debt so we can review it after. Thanks

Reply:
What do u mean about 

> happens in all tables

How do u test it? Could you provide more context, it'll be useful to create the tickets.

Reply:
I fixed it, it has to be inside drop down component the modal